Developer-Guide.md

```diff
@@ -273,22 +291,199 @@ $ popd
+```
+
+Restart CRI-O to take changes into account
+```
+$ sudo systemctl restart crio
+```
+
+### CRI-containerd
+
+If you picked CRI-containerd, please follow the instructions
+[here](https://github.com/containerd/cri#getting-started-for-developers)
+to get it properly installed.
+
+And for the customization part so that CRI-containerd will pick Kata Containers
+runtime, please follow our internal documentation
+[here](https://github.com/kata-containers/documentation/blob/master/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md#configure-containerd-to-use-kata-containers).
+
+## Install Kubernetes
+
+Depending on what your needs are and what you expect to do with Kubernetes,
+please refer to the following
+[documentation](https://kubernetes.io/docs/setup/pick-right-solution/) to
+install it correctly.
+
+Kubernetes talks with CRI implementations through a `container-runtime-endpoint`,
+also called CRI socket. This socket path is different depending on which CRI
+implementation you choosed, and the kubelet service has to be updated accordingly.
+
+### Configure for CRI-O
+
+_/etc/systemd/system/kubelet.service.d/0-crio.conf_
+```
+[Service]                                                 
+Environment="KUBELET_EXTRA_ARGS=--container-runtime=remote --runtime-request-timeout=15m --container-runtime-endpoint=unix:///var/run/crio/crio.sock"
+```
+
+### Configure for CRI-containerd
+
+_/etc/systemd/system/kubelet.service.d/0-cri-containerd.conf_
+```
+[Service]                                                 
+Environment="KUBELET_EXTRA_ARGS=--container-runtime=remote --runtime-request-timeout=15m --container-runtime-endpoint=unix:///run/containerd/containerd.sock"
+```
+For more details about CRI-containerd, click
+[here](https://github.com/kata-containers/documentation/blob/master/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md#configure-kubelet-to-use-containerd).
+
+## Run a Kubernetes pod with Kata Containers
+
+After you have updated your kubelet service based on the CRI implementation you
+are using, reload and restart kubelet, and start your cluster:
+```bash
+$ sudo systemctl daemon-reload
+$ sudo systemctl restart kubelet
+
+# If using CRI-O
+$ sudo kubeadm init --skip-preflight-checks --cri-socket /var/run/crio/crio.sock --pod-network-cidr=10.244.0.0/16
+
+# If using CRI-containerd
+$ sudo kubeadm init --skip-preflight-checks --cri-socket /run/containerd/containerd.sock --pod-network-cidr=10.244.0.0/16
+
+$ export KUBECONFIG=/etc/kubernetes/admin.conf
+```
+
+You can force kubelet to use Kata Containers by adding some _untrusted_
+annotation to your pod configuration. In our case, this will ensure Kata
+Containers is the selected runtime to run the described workload.
+
+_nginx-untrusted.yaml_
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx-untrusted
+  annotations:
+    io.kubernetes.cri.untrusted-workload: "true"
+spec:
+  containers:
+    name: nginx
+    image: nginx
+```
 
+And you simply need to run your pod:
```

Comment:
Suggested rewrite: Next, you run your pod:

Reply:
done